finagle-backend/src/main/scala/sttp/client/finagle/FinagleBackend.scala

```diff
@@ -0,0 +1,165 @@
+package sttp.client.finagle
+
+import com.twitter.finagle.Http.Client
+import com.twitter.finagle.{Http, http}
+import sttp.client.{BasicRequestBody, ByteArrayBody, ByteBufferBody, FileBody, FollowRedirectsBackend, IgnoreResponse, InputStreamBody, MappedResponseAs, MultipartBody, NoBody, NothingT, Request, Response, ResponseAs, ResponseAsByteArray, ResponseAsFile, ResponseAsFromMetadata, ResponseAsStream, ResponseMetadata, StringBody, SttpBackend}
+import com.twitter.util.{Try, Future => TFuture}
+import sttp.client.monad.MonadError
+import sttp.client.ws.WebSocketResponse
+import com.twitter.finagle.http.{FileElement, FormElement, RequestBuilder, SimpleElement, Method => FMethod, Response => FResponse}
+import com.twitter.io.Buf
+import com.twitter.io.Buf.{ByteArray, ByteBuffer}
+import com.twitter.util
+import sttp.client.internal.FileHelpers
+import sttp.model.{Header, Method, Part, StatusCode}
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.util.Duration
+import scala.io.Source
+
+class FinagleBackend(client: Client) extends SttpBackend[TFuture, Nothing, NothingT] {
+  override def send[T](request: Request[T, Nothing]): TFuture[Response[T]] = {
+    val service = client.newService(s"${request.uri.host}:${request.uri.port.getOrElse(80)}")
+    val finagleRequest = requestBodyToFinagle(request)
+    service.apply(finagleRequest).within(Duration.fromMilliseconds(request.options.readTimeout.toMillis))(DefaultTimer).flatMap{
+      fResponse =>
+        val code = StatusCode.unsafeApply(fResponse.statusCode)
+        val headers = fResponse.headerMap.map(h => Header.notValidated(h._1, h._2)).toList
+        val statusText = fResponse.status.reason
+        val responseMetadata = ResponseMetadata(headers, code, statusText)
+        val body = fromFinagleResponse(request.response, fResponse, responseMetadata)
+        service.close().flatMap(_ => body.map(sttp.client.Response(_, code, statusText, headers, Nil)))
+    }
+  }
+
+  /**
+    * Opens a websocket, using the given backend-specific handler.
+    *
+    * If the connection doesn't result in a websocket being opened, a failed effect is
+    * returned, or an exception is thrown (depending on `F`).
+    */
+  override def openWebsocket[T, WS_RESULT](request: Request[T, Nothing], handler: NothingT[WS_RESULT]): TFuture[WebSocketResponse[WS_RESULT]] = handler
+
+  override def close(): TFuture[Unit] = TFuture.Done
+
+  /**
+    * The effect wrapper for responses. Allows writing wrapper backends, which map/flatMap over
+    * the return value of [[send]] and [[openWebsocket]].
+    */
+  override def responseMonad: MonadError[TFuture] = new MonadError[TFuture] {
```

Comment:
it's also often worth overriding `eval` which can have a more efficient "wrapper-native"  implementation in some cases

Reply:
done